client/src/pages/MainPage.jsx

```diff
@@ -8,6 +8,8 @@ import "../styles/MainPage.css";
 
 const MainPage = () => {
 
+    const defaultEvents = useRef([]);
+
     const [selectedTab, setSelectedTab] = useState(Tab.EVENTS);
     const [events, setEvents] = useState([]);
```

Comment:
Instead of managing a ref defaultEvents and a separate state variable for events you can rely on useMemo

```
const [allEvents, setAllEvents] = useState([]);
const [statusFilter, setStatusFilter] = useState();
const displayedEvents = useMemo(() => defaultEvents.current.filter(event => 
            event.status === filter
        ), [allEvents, statusFilter]);
```